### PR TITLE
Fixation & making the changes in Common-utils.

### PR DESCRIPTION
**Pull request by amolnin**
Added required mocking.
Removed redundant tests.
Fixed the names for tests as per convention.

### CODE REVIEW

File: common-util/src/test/java/org/sunbird/common/models/AppTest.java [hunk 1]
```diff
@@ -2,192 +2,100 @@
...
-
-    return map;
-  }
+public class AppTest extends BaseForHttpTest {
+	private static final String data = "{\"request\": { \"search\": {\"contentType\": [\"Story\"] }}}";
+	private static Map<String, String> headers = new HashMap<String, String>();
+
+	@BeforeClass
+	public static void init() {
+		headers.put("content-type", "application/json");
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Code changes are not properly formatted using scripts/setup.sh. Get it done with help from Manzarul.

Comment (bvinayakumar) [reply common-util/src/test/java/org/sunbird/common/models/AppTest.java]: This applies to all Java files in this PR.

File: common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  OutputStreamWriter.class,
+  URL.class,
+  BufferedReader.class,
+  HttpUtil.class,
+  HttpClients.class,
+  Notification.class,
+  KeyCloakConnectionProvider.class,
+  KeyCloakServiceImpl.class
+})
+public abstract class BaseForHttpTest {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java]: Rename to BaseHttpTest

File: common-util/src/test/java/org/sunbird/common/models/AppTest.java [hunk 3]
```diff
@@ -97,97 +94,14 @@ public void testPatchMatch() {
...
   @Test
-  public void testEmailValidation() {
+  public void testEmailValidationSuccess() {
     boolean bool = ProjectUtil.isEmailvalid("amit.kumar@tarento.com");
     assertTrue(bool);
   }
 
   @Test
-  public void testEmailFailureValidation() {
+  public void testEmailFailureWithWrongFormat() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Should be part of ProjectUtilTest.

File: common-util/src/test/java/org/sunbird/common/models/AppTest.java [hunk 4]
```diff
@@ -97,97 +94,14 @@ public void testPatchMatch() {
   }
 
   @Test
-  public void testEmailValidation() {
+  public void testEmailValidationSuccess() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Should be part of ProjectUtilTest.

File: common-util/src/test/java/org/sunbird/common/models/AppTest.java [hunk 5]
```diff
@@ -37,7 +35,7 @@ public static void init() {
   }
 
   @Test
-  public void testGetResourceMethod() throws Exception {
+  public void testGetResourceSuccess() throws Exception {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: testGetResourceSuccess, testPostResourceSuccess, testPostResourceFailureWithWrongUrl and testPatchRequestSuccess should be part of HttpUtilTest.

File: common-util/src/test/java/org/sunbird/common/models/AppTest.java (hunk 4 above)
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Should be part of ProjectUtilTest.

File: common-util/src/test/java/org/sunbird/common/models/AppTest.java (hunk 3 above)
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Should be part of ProjectUtilTest.

File: common-util/src/test/java/org/sunbird/common/models/AppTest.java (hunk 4 above)
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Should be part of ProjectUtilTest.

Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Should be part of ProjectUtilTest.

Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Should be part of ProjectUtilTest.

Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Should be part of ProjectUtilTest.

[the remark this answers is not in the document]
Comment (bvinayakumar) [reply common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Rename to testIsEmailValidSuccess.

File: common-util/src/test/java/org/sunbird/common/models/AppTest.java (hunk 3 above)
Comment (bvinayakumar) [reply common-util/src/test/java/org/sunbird/common/models/AppTest.java]: Rename to testIsEmailValidFailureWithInvalidFormat.

File: common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    httpRules("/content/wrong/v3/list", null, true, null);
+    httpRules("v1/issuer/issuers", "{\"message\":\"success\"}");
+    httpRules("https://dev.ekstep.in/api/data/v3", "{\"message\":\"success\"}");
+  }
+
+  protected void httpRules(String urlContains, String outputexpected) {
+    httpRules(urlContains, outputexpected, false, null);
+  }
+
+  protected void httpRules(
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java]: Rename httpRules to mockHttpUrlResponse.

File: common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      String urlContains, String outputexpected, boolean throwError, String paramContains) {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java]: Rename outputexpected to outputExpected.

Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java]: Is paramContains used? In all cases, it seems to be only passed null value.

File: common-util/src/test/java/org/sunbird/common/models/util/HttpUtilTest.java [hunk 8]
```diff
@@ -23,24 +15,15 @@
...
-@RunWith(PowerMockRunner.class)
-@PrepareForTest(HttpClients.class)
-@PowerMockIgnore({"javax.management.*", "javax.net.ssl.*", "javax.security.*"})
-public class HttpUtilTest {
+public class HttpUtilTest extends BaseForHttpTest {
   public static final String JSON_STRING_DATA = "asdasasfasfsdfdsfdsfgsd";
 
   @Test
-  public void testpostFormData() {
+  public void testPostFormSuccess() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/util/HttpUtilTest.java]: Check naming of tests. e.g. testPostFormSuccess to testPostFormDataSuccess

File: common-util/src/test/java/org/sunbird/common/models/util/ProjectUtilTest.java [hunk 9]
```diff
@@ -15,19 +15,18 @@
...
 
 /** Created by arvind on 6/10/17. */
-public class ProjectUtilTest {
+public class ProjectUtilTest extends BaseForHttpTest {
 
   private PropertiesCache propertiesCache = ProjectUtil.propertiesCache;
 
   @Test
-  public void testMailTemplateContextNameAsent() {
+  public void testMailTemplateFailureWithNameAbsent() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/util/ProjectUtilTest.java]: Check naming of tests. e.g. testMailTemplateFailureWithNameAbsent to testGetContextFailureWithoutName

File: common-util/src/test/java/org/sunbird/common/models/util/fcm/FCMNotificationTest.java [hunk 10]
```diff
@@ -19,12 +37,20 @@
...
+  Notification.class
+})
+@PowerMockIgnore({"javax.management.*", "javax.net.ssl.*", "javax.security.*"})
 public class FCMNotificationTest {
 
- 
-  @Ignore
   @Test
-  public void successNotificationWithStringAndObjectData() {
+  public void testSendNotificationSuccessWithListAndStrings() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/util/fcm/FCMNotificationTest.java]: testSendNotificationSuccessWithListAndStringData

File: common-util/src/test/java/org/sunbird/common/models/util/fcm/FCMNotificationTest.java [hunk 11]
```diff
@@ -60,18 +86,44 @@ public void failureNotificationWithEmptyFCMURL() {
   }
 
   @Test
-  public void failureNotificationWithWrongJson() {
+  public void testSendNotificationFailureWithNullMap() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/models/util/fcm/FCMNotificationTest.java]: testSendNotificationFailureWithNullData

File: common-util/src/test/java/org/sunbird/common/request/CommonRequestValidatorTest.java [hunk 12]
```diff
@@ -19,7 +21,7 @@
 public class CommonRequestValidatorTest {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/CommonRequestValidatorTest.java]: Rename class to RequestValidatorTest.

File: common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java [hunk 13]
```diff
@@ -108,7 +108,7 @@ public void validateChangePasswordWithMissingNewPass() {
   }
 
   @Test
-  public void validateChangePasswordWithSameOldAndNewPass() {
+  public void testValidateChangePasswordFailureWithOldNewSame() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: testValidateChangePasswordFailureWithSameOldPassword

File: common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java [hunk 14]
```diff
@@ -502,13 +473,13 @@ public void validateAssignRoleWithUserId() {
   }
 
   @Test
-  @Ignore
-  public void validateAssignRoleWithProviderAndExternalId() {
+  public void testValidateUploadUserSuccessWithProviderAndExternalId() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: testValidateAssignRoleSuccessWithProviderAndExternalId

File: common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java [hunk 15]
```diff
@@ -535,7 +506,7 @@ public void validateAssignRoleWithProviderAndExternalId() {
    */
 
   @Test
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: Remove commented test phoneAndEmailValidationForUpdateUser.

File: common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java [hunk 16]
```diff
@@ -623,22 +594,21 @@ public void validateWebPagesTestWhenNull() {
   }
 
   @Test
-  @Ignore
-  public void doUserBasicValidationUserNameTest() {
+  public void testCreateUserValidationFailureWithEmptyFirstName() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: Rename testCreateUserValidation* tests to testCreateUserBasicValidation*

File: common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java [hunk 17]
```diff
@@ -482,12 +451,14 @@ public void validateUploadUserWithProviderAndExternalId() {
   }
 
   @Test
-  @Ignore
-  public void validateAssignRoleWithUserId() {
+  public void testValidateAssignRoleSuccess() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: Move all user request validator tests to UserRequestValidatorTest class.

File: common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java (hunk 7 above)
Comment (amolnin) [reply common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java]: Used from sub-class only - current methods usage defaults to null

**Pull request merged**